src/sinks/appsignal/integration_tests.rs

```diff
@@ -1,52 +1,78 @@
+use vector_core::event::{
+    BatchNotifier, BatchStatus, Event, LogEvent, Metric, MetricKind, MetricValue,
+};
 
 use crate::{
     config::SinkConfig,
-    sinks::appsignal::AppsignalSinkConfig,
-    sinks::util::test::load_sink,
+    sinks::appsignal::AppsignalConfig,
+    sinks::util::test::{build_test_server_status, load_sink},
     test_util::{
         components::{
             assert_sink_compliance, assert_sink_error, run_and_assert_sink_compliance,
-            COMPONENT_ERROR_TAGS, SINK_TAGS,
+            COMPONENT_ERROR_TAGS, HTTP_SINK_TAGS,
         },
-        generate_lines_with_stream, map_event_batch_stream,
+        generate_lines_with_stream, map_event_batch_stream, next_addr,
     },
 };
 
+async fn start_test(events: Vec<Event>) -> (Vec<Event>, Receiver<(http::request::Parts, Bytes)>) {
+    let config = indoc! {r#"
+        push_api_key = "${TEST_APPSIGNAL_PUSH_API_KEY}"
+        compression = "none"
+    "#};
+    let config = config.replace("${TEST_APPSIGNAL_PUSH_API_KEY}", &push_api_key());
+    let (mut config, cx) = load_sink::<AppsignalConfig>(config.as_str()).unwrap();
+    let addr = next_addr();
+    // Set the endpoint to a local server so we can fetch the sent events later
+    config.endpoint = format!("http://{}", addr);
+
+    let (sink, _) = config.build(cx).await.unwrap();
+
+    // Always return OK from server. We're not testing responses.
+    let (rx, _trigger, server) = build_test_server_status(addr, StatusCode::OK);
+    tokio::spawn(server);
+
+    let (batch, receiver) = BatchNotifier::new_with_receiver();
+
+    let stream = map_event_batch_stream(stream::iter(events.clone()), Some(batch));
+
+    sink.run(stream).await.unwrap();
+    assert_eq!(receiver.await, BatchStatus::Delivered);
+
+    (events, rx)
+}
+
 #[tokio::test]
 async fn logs_real_endpoint() {
     let config = indoc! {r#"
         push_api_key = "${TEST_APPSIGNAL_PUSH_API_KEY}"
     "#};
-    let api_key = std::env::var("TEST_APPSIGNAL_PUSH_API_KEY")
-        .expect("couldn't find the AppSignal push API key in environment variables");
-    assert!(!api_key.is_empty(), "$TEST_APPSIGNAL_PUSH_API_KEY required");
-    let config = config.replace("${TEST_APPSIGNAL_PUSH_API_KEY}", &api_key);
-    let (config, cx) = load_sink::<AppsignalSinkConfig>(config.as_str()).unwrap();
+    let config = config.replace("${TEST_APPSIGNAL_PUSH_API_KEY}", &push_api_key());
+    let (config, cx) = load_sink::<AppsignalConfig>(config.as_str()).unwrap();
 
     let (sink, _) = config.build(cx).await.unwrap();
     let (batch, receiver) = BatchNotifier::new_with_receiver();
     let generator = |index| format!("this is a log with index {}", index);
     let (_, events) = generate_lines_with_stream(generator, 10, Some(batch));
 
-    run_and_assert_sink_compliance(sink, events, &SINK_TAGS).await;
+    run_and_assert_sink_compliance(sink, events, &HTTP_SINK_TAGS).await;
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review (if it exists :^)  -->
**👏 praise:** ‏Thanks for adjusting to use the `HTTP_SINK_TAGS`. It covers the additional property of `endpoint` .

We have some work in flight to better handle the validation of these types of internal metrics.